src/nanoarrow/array_test.cc

```diff
@@ -87,12 +87,6 @@ TEST(ArrayTest, ArrayTestAllocateChildren) {
   EXPECT_EQ(array.n_children, 0);
   ArrowArrayRelease(&array);
 
-  ASSERT_EQ(ArrowArrayInitFromType(&array, NANOARROW_TYPE_STRUCT), NANOARROW_OK);
```

Comment:
Unfortunately ASAN does not like this code block:

```
==897112==ERROR: AddressSanitizer: requested allocation size 0x7ffffffffffffff8 (0x8000000000000ff8 after adjustments for alignment, red zones etc.) exceeds maximum supported size of 0x10000000000 (thread T0)
```

It might be possible to add a function attribute to disable instrumentation:

https://github.com/google/sanitizers/wiki/AddressSanitizer#turning-off-instrumentation

But couldn't find a reasonable way to make that work with gtest

Reply:
actually solved via `ASAN_OPTIONS=allocator_may_return_null=1`

Reply:
Meson has allows you to pick either `address,undefined` or `leak` (amongst other options) but not both at the same time. We may want to consider a second compilation for leak checking in that case